Extension/src/LanguageServer/extension.ts

```diff
@@ -770,64 +769,66 @@ async function suggestInsidersChannel(): Promise<void> {
     }
 }
 
-function applyUpdate(buildInfo: BuildInfo): Promise<void> {
-    return new Promise<void>((resolve, reject) => {
-        tmp.file({postfix: '.vsix'}, async (err, vsixPath, fd, cleanupCallback) => {
-            if (err) {
-                reject(new Error('Failed to create vsix file'));
-                return;
-            }
+async function applyUpdate(buildInfo: BuildInfo): Promise<void> {
+    const tempVSIX: any = await util.CreateTempFileWithPostfix('.vsix').catch(error => {
+        if (error.message.indexOf('/') !== -1 || error.message.indexOf('\\') !== -1) {
+            error.message = "Potential PII hidden";
+        }
+        telemetry.logLanguageServerEvent('installVsix', { 'error': error.message, 'success': 'false' });
+        return;
```

Comment:
I'm pretty sure that this just exits the `.catch`, not `applyUpdate`.  That's probably why you had to make `tempVSIX` an `any`.  If you use `try/catch` instead, you can exit within the `catch` block which is what I think you want to do.  If this was intentional, then remove the `return` because it is unnecessary.

Reply:
Actually, I think you can move the `util.createTempFileWithPostfix` call inside the bigger try/catch below and piggy back on the error handling that sends the telemetry since it's the same code.